### ISSUE TITLE
release 6.0?

### ISSUE THREAD

**Issue by AlJohri** [jupyter/jupyter_client#401]
[update] sorry for creating the empty issue, I pressed enter on accident

I'm getting this version conflict issue right now due to jupyter-client hardcoding the prompt-toolkit requirements:

- **prompt-toolkit [required: >=1.0.15,<2.0.0, installed: 1.0.15]**
- **prompt-toolkit [required: >=2.0.0,<2.1.0, installed: 1.0.15]**

```
Could not find a version that matches prompt_toolkit<2.1.0,==1.0.15,>=2.0.0
Tried: 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.8, 0.9, 0.10, 0.11, 0.12, 0.13, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19, 0.20, 0.21, 0.22, 0.23, 0.25, 0.26, 0.28, 0.30, 0.31, 0.32, 0.32, 0.32, 0.33, 0.33, 0.33, 0.34, 0.34, 0.34, 0.35, 0.35, 0.35, 0.36, 0.36, 0.36, 0.37, 0.37, 0.37, 0.38, 0.38, 0.38, 0.39, 0.39, 0.39, 0.40, 0.40, 0.40, 0.41, 0.41, 0.41, 0.42, 0.42, 0.42, 0.43, 0.43, 0.43, 0.44, 0.44, 0.44, 0.45, 0.45, 0.45, 0.46, 0.46, 0.46, 0.47, 0.47, 0.47, 0.48, 0.48, 0.48, 0.49, 0.49, 0.49, 0.50, 0.50, 0.50, 0.51, 0.51, 0.51, 0.52, 0.52, 0.52, 0.53, 0.53, 0.53, 0.54, 0.54, 0.54, 0.55, 0.55, 0.55, 0.56, 0.56, 0.56, 0.57, 0.57, 0.57, 0.58, 0.58, 0.58, 0.59, 0.59, 0.59, 0.60, 0.60, 0.60, 1.0.0, 1.0.0, 1.0.0, 1.0.1, 1.0.1, 1.0.1, 1.0.2, 1.0.2, 1.0.2, 1.0.3, 1.0.3, 1.0.3, 1.0.4, 1.0.4, 1.0.4, 1.0.5, 1.0.5, 1.0.5, 1.0.6, 1.0.6, 1.0.6, 1.0.7, 1.0.7, 1.0.7, 1.0.8, 1.0.8, 1.0.8, 1.0.9, 1.0.9, 1.0.9, 1.0.10, 1.0.10, 1.0.10, 1.0.13, 1.0.13, 1.0.13, 1.0.14, 1.0.14, 1.0.14, 1.0.14, 1.0.15, 1.0.15, 1.0.15, 1.0.15, 2.0.1, 2.0.1, 2.0.1, 2.0.2, 2.0.2, 2.0.2, 2.0.3, 2.0.3, 2.0.3, 2.0.4, 2.0.4, 2.0.4, 2.0.5, 2.0.5, 2.0.5, 2.0.6, 2.0.6, 2.0.6
There are incompatible versions in the resolved dependencies.
```

```
- jupyter-console [required: Any, installed: 6.0.0]
  - ipykernel [required: Any, installed: 5.1.0]
    - ipython [required: >=5.0.0, installed: 6.4.0]
      - appnope [required: Any, installed: 0.1.0]
      - backcall [required: Any, installed: 0.1.0]
      - decorator [required: Any, installed: 4.3.0]
      - jedi [required: >=0.10, installed: 0.12.0]
        - parso [required: >=0.2.0, installed: 0.2.1]
      - pexpect [required: Any, installed: 4.6.0]
        - ptyprocess [required: >=0.5, installed: 0.5.2]
      - pickleshare [required: Any, installed: 0.7.4]
      - prompt-toolkit [required: >=1.0.15,<2.0.0, installed: 1.0.15]
        - six [required: >=1.9.0, installed: 1.11.0]
        - wcwidth [required: Any, installed: 0.1.7]
      - pygments [required: Any, installed: 2.2.0]
      - setuptools [required: >=18.5, installed: 40.0.0]
      - simplegeneric [required: >0.8, installed: 0.8.1]
      - traitlets [required: >=4.2, installed: 4.3.2]
        - decorator [required: Any, installed: 4.3.0]
        - ipython-genutils [required: Any, installed: 0.2.0]
        - six [required: Any, installed: 1.11.0]
    - jupyter-client [required: Any, installed: 5.2.3]
      - jupyter-core [required: Any, installed: 4.4.0]
        - traitlets [required: Any, installed: 4.3.2]
          - decorator [required: Any, installed: 4.3.0]
          - ipython-genutils [required: Any, installed: 0.2.0]
          - six [required: Any, installed: 1.11.0]
      - python-dateutil [required: >=2.1, installed: 2.7.3]
        - six [required: >=1.5, installed: 1.11.0]
      - pyzmq [required: >=13, installed: 17.1.2]
      - tornado [required: >=4.1, installed: 5.0.2]
      - traitlets [required: Any, installed: 4.3.2]
        - decorator [required: Any, installed: 4.3.0]
        - ipython-genutils [required: Any, installed: 0.2.0]
        - six [required: Any, installed: 1.11.0]
    - tornado [required: >=4.2, installed: 5.0.2]
    - traitlets [required: >=4.1.0, installed: 4.3.2]
      - decorator [required: Any, installed: 4.3.0]
      - ipython-genutils [required: Any, installed: 0.2.0]
      - six [required: Any, installed: 1.11.0]
  - ipython [required: Any, installed: 6.4.0]
    - appnope [required: Any, installed: 0.1.0]
    - backcall [required: Any, installed: 0.1.0]
    - decorator [required: Any, installed: 4.3.0]
    - jedi [required: >=0.10, installed: 0.12.0]
      - parso [required: >=0.2.0, installed: 0.2.1]
    - pexpect [required: Any, installed: 4.6.0]
      - ptyprocess [required: >=0.5, installed: 0.5.2]
    - pickleshare [required: Any, installed: 0.7.4]
    - prompt-toolkit [required: >=1.0.15,<2.0.0, installed: 1.0.15]
      - six [required: >=1.9.0, installed: 1.11.0]
      - wcwidth [required: Any, installed: 0.1.7]
    - pygments [required: Any, installed: 2.2.0]
    - setuptools [required: >=18.5, installed: 40.0.0]
    - simplegeneric [required: >0.8, installed: 0.8.1]
    - traitlets [required: >=4.2, installed: 4.3.2]
      - decorator [required: Any, installed: 4.3.0]
      - ipython-genutils [required: Any, installed: 0.2.0]
      - six [required: Any, installed: 1.11.0]
  - jupyter-client [required: Any, installed: 5.2.3]
    - jupyter-core [required: Any, installed: 4.4.0]
      - traitlets [required: Any, installed: 4.3.2]
        - decorator [required: Any, installed: 4.3.0]
        - ipython-genutils [required: Any, installed: 0.2.0]
        - six [required: Any, installed: 1.11.0]
    - python-dateutil [required: >=2.1, installed: 2.7.3]
      - six [required: >=1.5, installed: 1.11.0]
    - pyzmq [required: >=13, installed: 17.1.2]
    - tornado [required: >=4.1, installed: 5.0.2]
    - traitlets [required: Any, installed: 4.3.2]
      - decorator [required: Any, installed: 4.3.0]
      - ipython-genutils [required: Any, installed: 0.2.0]
      - six [required: Any, installed: 1.11.0]
  - prompt-toolkit [required: >=2.0.0,<2.1.0, installed: 1.0.15]
    - six [required: >=1.9.0, installed: 1.11.0]
    - wcwidth [required: Any, installed: 0.1.7]
  - pygments [required: Any, installed: 2.2.0]
```

**Comment by AlJohri** [comment]
via pip
```
jupyter-console 6.0.0 has requirement prompt-toolkit<2.1.0,>=2.0.0, but you'll have prompt-toolkit 1.0.15 which is incompatible.
```

**Comment by minrk** [comment]
jupyter-client doesn't use prompt-toolkit at all. Perhaps you meant jupyter-console? That has released 6.0. I think latest of everything works with prompt-toolkit 2.0 now
